code/modules/awaymissions/corpse.dm

```diff
@@ -4,294 +4,474 @@
-	var/mob/living/carbon/human/human/M = new /mob/living/carbon/human/human (src.loc)
-	M.real_name = src.name
-	M.death(1) //Kills the new mob
+/obj/effect/mob_spawn/attack_ghost(mob/user)
+	if(ticker.current_state != GAME_STATE_PLAYING || !loc)
+		return
+	if(!uses)
+		to_chat(user, "<span class='warning'>This spawner is out of charges!</span>")
+		return
+	if(jobban_isbanned(user, banType))
+		to_chat(user, "<span class='warning'>You are jobanned!</span>")
+		return
+	var/ghost_role = alert("Become [mob_name]? (Warning, You can no longer be cloned!)",,"Yes","No")
+	if(ghost_role == "No" || !loc)
+		return
+	log_game("[user.ckey] became [mob_name]")
+	create(ckey = user.ckey)
+
+/obj/effect/mob_spawn/New()
+	. = ..()
+	if(ticker && ticker.current_state > GAME_STATE_PLAYING)
+		create()
+	else
+		poi_list |= src
+		LAZYADD(mob_spawners[name], src)
+
+/obj/effect/mob_spawn/Destroy()
+	poi_list -= src
+	var/list/spawners = mob_spawners[name]
+	LAZYREMOVE(spawners, src)
+	return ..()
+
+/obj/effect/mob_spawn/proc/special(mob/M)
+	return
+
+/obj/effect/mob_spawn/proc/equip(mob/M)
+	return
+
+/obj/effect/mob_spawn/proc/create(ckey, flavour = TRUE, name)
+	var/mob/living/M = new mob_type(get_turf(src)) //living mobs only
+	if(!random)
+		M.real_name = mob_name ? mob_name : M.name
+		if(!mob_gender)
+			mob_gender = pick(MALE, FEMALE)
+		M.gender = mob_gender
+	if(faction)
+		M.faction = list(faction)
+	if(disease)
+		M.ForceContractDisease(new disease)
+	if(death)
+		M.death(1) //Kills the new mob
+
 	M.adjustOxyLoss(oxy_damage)
 	M.adjustBruteLoss(brute_damage)
-	M.timeofdeath = timeofdeath
+	M.adjustFireLoss(burn_damage)
+	M.color = mob_color
+	equip(M)
+
+	if(ckey)
+		M.ckey = ckey
+		if(flavour)
+			to_chat(M, "[flavour_text]")
+		var/datum/mind/MM = M.mind
+		if(objectives)
+			for(var/objective in objectives)
+				MM.objectives += new/datum/objective(objective)
+		if(assignedrole)
+			M.mind.assigned_role = assignedrole
+		special(M, name)
+		MM.name = M.real_name
+	if(uses > 0)
+		uses--
+	if(!permanent && !uses)
+		qdel(src)
+
+// Base version - place these on maps/templates.
+/obj/effect/mob_spawn/human
+	mob_type = /mob/living/carbon/human
+	//Human specific stuff.
+	var/mob_species = null		//Set to make them a mutant race such as lizard or skeleton. Uses the datum typepath instead of the ID.
+	var/datum/outfit/outfit = /datum/outfit	//If this is a path, it will be instanced in Initialize()
+	var/disable_pda = TRUE
+	//All of these only affect the ID that the outfit has placed in the ID slot
+	var/id_job = null			//Such as "Clown" or "Chef." This just determines what the ID reads as, not their access
+	var/id_access = null		//This is for access. See access.dm for which jobs give what access. Use "Captain" if you want it to be all access.
+	var/id_access_list = null	//Allows you to manually add access to an ID card.
+	assignedrole = "Ghost Role"

```

Comment:
Maybe "Spawned"+name_of_type? "Ghost" would cause them to show up when you search for ghosts in PPN.

Reply:
noted